### PR TITLE
v23.2.0-alpha.1 release notes

### PR DESCRIPTION
**Pull request by ianjevans**
Fixes DOC-7782.

#### Major
[releases/v23.2.md](https://deploy-preview-17890--cockroachdb-docs.netlify.app/docs/releases/v23.2.html)

#### Minor
[releases/v22.2.md](https://deploy-preview-17890--cockroachdb-docs.netlify.app/docs/releases/v22.2.html)
[releases/v23.1.md](https://deploy-preview-17890--cockroachdb-docs.netlify.app/docs/releases/v23.1.html)

### CODE REVIEW

File: src/current/_data/releases.yml
Comment (rmloveland) [inline src/current/_data/releases.yml:4871]: @ianjevans I think this is a typo during the RN script invocation

I think this should read v23.2.0-alpha.1

File: src/current/_includes/releases/v23.2/v23.2.0-alpha1.md
Comment (rmloveland) [inline src/current/_includes/releases/v23.2/v23.2.0-alpha1.md:66]: typo: CockroachDB

File: src/current/_includes/releases/v23.2/v23.2.0-alpha1.md
Comment (rmloveland) [inline src/current/_includes/releases/v23.2/v23.2.0-alpha1.md:75]: Link to `CREATE` is missing a closing paren, fails to render

File: src/current/_includes/releases/v23.2/v23.2.0-alpha1.md
Comment (rmloveland) [inline src/current/_includes/releases/v23.2/v23.2.0-alpha1.md:167]: link fails to render due to missing close paren

File: src/current/_includes/releases/v23.2/v23.2.0-alpha1.md
Comment (rmloveland) [inline src/current/_includes/releases/v23.2/v23.2.0-alpha1.md:455]: I would delete this section starting at "We would like to thank ..." . I know at least some of these folks are CRL.  This happens b/c they have a personal email in their `.git/config` or similar

**Review by rmloveland — APPROVED** [review]
LGTM, found an issue with the name and some link renders.  Should be good to merge once those are set

**Comment by mdlinville** [comment]
@kathancox @rmloveland @celiala PTAL and make sure the latest commit looks right for updated SHA.

**Comment by mdlinville** [comment]
@nickvigilante Can you help with the Netlify error? I think it's breaking a link because an alpha.1 has no `{{previous_version}}` set (I verified that none of the previous alpha.1 did either). Please advise.

File: src/current/_data/versions.csv
Comment (nickvigilante) [inline src/current/_data/versions.csv:15]: @mdlinville It seems like this row was accidentally deleted. Not sure how that happened. I'll follow up with Ian when he gets back into the office to see if he observed that with the `release-notes.py` script. I don't believe the script does anything to delete rows in `versions.csv`, but I reserve the right to be wrong.

**Comment by kathancox** [comment]
@mdlinville Do Rich and I need to review the whole set of release notes or the final commit?

File: src/current/_data/versions.csv
Comment (mdlinville) [reply src/current/_data/versions.csv:15]: I think Ian intentionally removed it based on https://cockroachlabs.slack.com/archives/G01Q35D80EM/p1694729513545189?thread_ts=1694728082.428199&cid=G01Q35D80EM. Should I try putting it back, and revert the changes to the k8s file?

Comment (ianjevans) [reply src/current/_data/versions.csv:15]: I removed this line in order to get `release_notes.py` to work, but I *thought* I reverted the change before pushing.

File: src/current/_includes/releases/v23.2/v23.2.0-alpha.1.md
Comment (kathancox) [inline src/current/_includes/releases/v23.2/v23.2.0-alpha.1.md:161]: Why are we using the full link here? prior to the liquid? (Sorry if I missed something about this.)

File: src/current/_includes/releases/v23.2/v23.2.0-alpha.1.md
Comment (kathancox) [inline src/current/_includes/releases/v23.2/v23.2.0-alpha.1.md:167]: There's an additional bullet point here that is throwing off the rendering.

File: src/current/_includes/releases/v23.2/v23.2.0-alpha.1.md
Comment (kathancox) [inline src/current/_includes/releases/v23.2/v23.2.0-alpha.1.md:161]: code formatting for `READ COMMITTED` and `false`

File: src/current/_includes/releases/v23.2/v23.2.0-alpha.1.md
Comment (kathancox) [inline src/current/_includes/releases/v23.2/v23.2.0-alpha.1.md:171]: Space missing after the period following "`@hourly`"

Comment (kathancox) [reply src/current/_includes/releases/v23.2/v23.2.0-alpha.1.md:171]: Correction to typo:

> As part of this change, the output of the `SHOW CREATE TABLE` statements now includes the `ttl_cron_job` parameter only if it is explicitly set.

Comment (kathancox) [reply src/current/_includes/releases/v23.2/v23.2.0-alpha.1.md:171]: And maybe link the `ttl_job_cron` to `row-level-ttl.md`?

File: src/current/_includes/releases/v23.2/v23.2.0-alpha.1.md
Comment (kathancox) [inline src/current/_includes/releases/v23.2/v23.2.0-alpha.1.md:185]: This isn't quite rendering properly. I think the "For accurate..." sentence needs to bump in a tab one more. and the same for the points following

Comment (kathancox) [reply src/current/_includes/releases/v23.2/v23.2.0-alpha.1.md:185]: It is affecting the release notes that follow this one too.

File: src/current/_includes/releases/v23.2/v23.2.0-alpha.1.md
Comment (kathancox) [inline src/current/_includes/releases/v23.2/v23.2.0-alpha.1.md:384]: Maybe in quotes, or maybe in bold, to make it easier to read that.

>  a "no data empty graph" state

File: src/current/_data/versions.csv
Comment (nickvigilante) [reply src/current/_data/versions.csv:15]: Gotcha. That bug should be fixed, so yes, if someone can just make sure this line gets added back in, we should be golden, and the change in `src/current/v23.2/orchestrate-cockroachdb-with-kubernetes-multi-cluster.md` can (and should) be reverted.

File: src/current/_includes/releases/v23.2/v23.2.0-alpha.1.md
Comment (kathancox) [inline src/current/_includes/releases/v23.2/v23.2.0-alpha.1.md:474]: The second sentence here, I would maybe break up:

> The fix only applies to future schema changes. To process existing stuck jobs, manually force-enqueue the relevant ranges in the MVCC GC queue from the DB Console's [Advanced Debug](https://www.cockroachlabs.com/docs/{{ page.version.version }}/ui-debug-pages) page.

File: src/current/_includes/releases/v23.2/v23.2.0-alpha.1.md
Comment (kathancox) [inline src/current/_includes/releases/v23.2/v23.2.0-alpha.1.md:480]: Missing period at end of the note here.

File: src/current/_includes/releases/v23.2/v23.2.0-alpha.1.md
Comment (kathancox) [inline src/current/_includes/releases/v23.2/v23.2.0-alpha.1.md:510]: Serial comma add here and inline code on `make`:

> The common targets `make [all]`, `make test`, and `make install` remain for compatibility with most UNIX installation guides. The previous `make` rules remain available via `make -C build/GNUmakefile.obsolete ...`.

File: src/current/_includes/releases/v23.2/v23.2.0-alpha.1.md
Comment (kathancox) [inline src/current/_includes/releases/v23.2/v23.2.0-alpha.1.md:971]: To delete?

**Review by kathancox — APPROVED** [review]
Hey! there are a few rendering issues with some of the notes, but other than that all nits. Sorry I couldn't add suggestions, let me know if anything is not clear.

File: src/current/_includes/releases/v23.2/v23.2.0-alpha.1.md
Comment (nickvigilante) [reply src/current/_includes/releases/v23.2/v23.2.0-alpha.1.md:161]: @kathancox The full link will make less work for me later on when we split the releases into its own Netlify site. We'll need to use full links between each docs page so that htmltest doesn't break.

Aside from that, the link will fail htmltest anyway, because of the Liquid. Proposing this instead:

```suggestion
- The new [cluster setting](https://www.cockroachlabs.com/docs/v23.2/cluster-settings) `sql.txn.read_committed_syntax.enabled`, controls whether transactions run under `READ COMMITTED` or `SERIALIZABLE` isolation. It defaults to false. When set to true, the following statements will configure transactions to run under READ COMMITTED isolation:
```

File: src/current/_includes/releases/v23.2/v23.2.0-alpha.1.md
Comment (nickvigilante) [inline src/current/_includes/releases/v23.2/v23.2.0-alpha.1.md:170]:
```suggestion
- The [cluster setting](https://www.cockroachlabs.com/docs/v23.2/cluster-settings) `sql.metrics.statement_details.gateway_node.enabled` now defaults to false, to reduce the number of rows generated in SQL Statistics pages. [#107788][#107788]
```

File: src/current/_includes/releases/v23.2/v23.2.0-alpha.1.md
Comment (nickvigilante) [reply src/current/_includes/releases/v23.2/v23.2.0-alpha.1.md:474]: Additionally, `s/{{ page.version.version }}/v23.2/g`.

**Review by nickvigilante — DISMISSED** [review]
`s/{{ page.version.version }}/v23.2/g`

I didn't review for content here, just for links.

File: src/current/_includes/releases/v23.2/v23.2.0-alpha.1.md
Comment (mdlinville) [inline src/current/_includes/releases/v23.2/v23.2.0-alpha.1.md:510]:
```suggestion
- The top-level `Makefile` was replaced by a stub `GNUmakefile` which defers its behavior to `dev`. The common targets `make [all]`, `make test`, and `make install` remain for compatibility with most UNIX installation guides. The previous `make` rules remain available via `make -C build/GNUmakefile.obsolete ...`. [#84565][#84565]
```

File: src/current/_includes/releases/v23.2/v23.2.0-alpha.1.md
Comment (mdlinville) [reply src/current/_includes/releases/v23.2/v23.2.0-alpha.1.md:971]: Yeah before merging. This is to help reviewers and @celiala to verify that I got all the relevant commits for the updated hash.

File: src/current/_includes/releases/v23.2/v23.2.0-alpha.1.md
Comment (celiala) [inline src/current/_includes/releases/v23.2/v23.2.0-alpha.1.md:188]: It's possible that this is how the commit should be mapped/formatted in the release notes?

But the notes for this one commit ([8f3e328a00a5d388a8cfd1664756c343e43a6d3b](https://github.com/cockroachdb/cockroach/pull/104417/commits/8f3e328a00a5d388a8cfd1664756c343e43a6d3b))

Is getting shifted left / having multiple additional bullets (instead of it being nested until the main bullet for the PR)

image

Comment (celiala) [reply src/current/_includes/releases/v23.2/v23.2.0-alpha.1.md:188]: update: okay, I see this happening for a few commits, so perhaps this is expected / fine.

**Review by celiala — COMMENTED** [review]
I didn't review the whole PR, but spot checked for a few PRs and confirmed that they're present in this alpha.1 release notes.

File: src/current/_includes/releases/v23.2/v23.2.0-alpha.1.md
Comment (celiala) [inline src/current/_includes/releases/v23.2/v23.2.0-alpha.1.md:479]: > without the `EXECUTE` privilege to execute a a user-defined function

nit: remove duplicate "a"

File: src/current/_includes/releases/v23.2/v23.2.0-alpha.1.md
Comment (celiala) [inline src/current/_includes/releases/v23.2/v23.2.0-alpha.1.md:146]: > Fixed an issue where the UI was

s/was/were/

File: src/current/_includes/releases/v23.2/v23.2.0-alpha.1.md
Comment (celiala) [inline src/current/_includes/releases/v23.2/v23.2.0-alpha.1.md:336]: > from taking to long

s/to/too/

File: src/current/_includes/releases/v23.2/v23.2.0-alpha.1.md
Comment (celiala) [inline src/current/_includes/releases/v23.2/v23.2.0-alpha.1.md:416]: s/ocurred/occurred/

File: src/current/_includes/releases/v23.2/v23.2.0-alpha.1.md
Comment (celiala) [inline src/current/_includes/releases/v23.2/v23.2.0-alpha.1.md:439]: > could sometime leave

s/sometime/sometimes/

File: src/current/_includes/releases/v23.2/v23.2.0-alpha.1.md
Comment (celiala) [inline src/current/_includes/releases/v23.2/v23.2.0-alpha.1.md:459]: s/statment/statement/

**Review by celiala — COMMENTED** [review]
(there are suuuuper nitty nits - feel free to ignore ;) )

File: src/current/_includes/releases/v23.2/v23.2.0-alpha.1.md
Comment (mdlinville) [reply src/current/_includes/releases/v23.2/v23.2.0-alpha.1.md]: There were some formatting issues but I think they are fixed now. PTAL again.

Comment (celiala) [reply src/current/_includes/releases/v23.2/v23.2.0-alpha.1.md]: LGTM - thanks!

File: src/current/_includes/releases/v23.2/v23.2.0-alpha.1.md
Comment (kathancox) [inline src/current/_includes/releases/v23.2/v23.2.0-alpha.1.md:16]:
```suggestion
- Users who have the [`CREATEROLE` role option]({% link v23.2/grant.md %}) can now grant and revoke role membership in any non-admin role.  This change also removes the [`sql.auth.createrole_allows_grant_role_membership.enabled` cluster setting]({% link v23.2/cluster-settings.md %}), which was added in v23.1. In v23.2, the cluster setting is effectively always true. [#104376][#104376]
```

**Review by kathancox — APPROVED** [review]

**Review by nickvigilante — APPROVED** [review]
A bunch of nits caught by Vale, but LGTM otherwise!

**Pull request merged**
